shoes-core/lib/shoes/color.rb

```diff
@@ -61,11 +61,7 @@ def black?
 
     def <=>(other)
       raise_class_mismatch_error(other) unless other.is_a?(self.class)
-      if same_base_color?(other)
-        @alpha <=> other.alpha
-      else
-        less_or_greater_than other
-      end
+      [red, green, blue, alpha] <=> [other.red, other.green, other.blue, other.alpha]
```

Comment:
Yep, it's definitely my overlook. It's hard to reason about color comparison, but if that breaks compatibility with other shoes I would rather stay away from merging that in.